setup/install/providers/aws/aws-ec2.md

```diff
@@ -102,47 +113,252 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
+
+### AWS IAM user or roles
+
+For IAM, you can either create IAM users or roles based on your requirements:
+
+1. Set Up AWS IAM structure and use an AWS IAM User *spinnakerManaging* with AccessKey and Secret. This option is useful for creating users with AWS Access Key and secret. This is a common configuration.
+2. Using AWS IAM Roles, such as creating a *ManagingRole* and *ManagedRoles*. This option might be needed for environments that have extra security considerations. The EC2 instance where Spinnaker is used to deploy AWS resources has the *ManagingRole* attached. The *ManagingRole* contains the AWS IAM policies required to manage AWS resources.
+
+### Create AWS IAM user
+
+1. Navigate to the AWS IAM console.
+2. Switch to the **spinnakerManaging** AWS Account.
+3. Add a user and name it **spinnakerManaging**.
+4. Check the “Programmatic access” checkbox.
+5. Add tags to help you identify this user, such as "spinnakerManaging".
+6. Create the user.
+7. Save the “Access key ID” and “Secret access key” somewhere secure, such as a secret management system like Vault, AWS Secrets manager, or 1Password.
+8. Save the AWS ARN for the user, for example: `arn:aws:iam::100000000001:user/spinnakerManaging`. You need the ARN later when you configure the managed accounts to trust the managing account (IAM user).
+
+### Create Roles
+
+#### Create Managed Roles in each target AWS Account
+
+First, create the **spinnakerManaged** role for the **develop ID=200000000002**. Then, repeat the same steps to create **spinnakerManaged** in **staging ID=300000000003**.
+
+1. Navigate to the AWS IAM console.
+2. Switch to the AWS Account you want to create the roll for.
+3. Go to **Roles > Create Role**.
+4. Select EC2. You can change this later, because we want to specify an explicit consumer of this role in a later stage.
+5. For permissions, search for "PowerUserAccess" and select this policy. This gives the role permission to access AWS services.
+6. Add tags that will help you identify this role.
+7. Enter a role name: spinnakerManaged.
+8. Create the roll.
+9. Select the created role and add an inline policy using the following JSON snippet: 
+
+   ```json
+   {
+       "Version": "2012-10-17",
+       "Statement": [
+           {
+               "Action": [
+                   "iam:ListServerCertificates",
+                   "iam:PassRole"
+               ],
+               "Resource": [
+                   "*"
+               ],
+               "Effect": "Allow"
+           }
+       ]
+   }
+   ```
+
+10. Name the policy "PassRole-and-Certificates".
+11. Create the policy.
+
+Repeat these steps for the second AWS environment. In this example, that is the **staging** environment.
+
+### Create the role BaseIAMRole
+
+BaseIAMRole is the default role that Spinnaker will use if the **spinnakerManaged** role is not found.
+
+1. Navigate to the AWS IAM console. 
+2. Create the role.
+3. Select **EC2**, and click **Next: Permissions**
```

Comment:
Is it intentional that the `BaseIAMRole` have no permission policies attached?

Reply:
Great Spotting. I don't give permissions to that role because is a default one. If something in cloud Driver fails and it requires the use of the default role then errors will be generated.
Perhaps it should have basic permissions to interact with AWS EC2 AMIs or some default policy but I don't think it should have Administration powers of any kind.

- What do you think? @brian-armory 

Actually the BaseIAM role only has a default permissions policy that allows assumeRole
```
  BaseIAMRole:
    Type: AWS::IAM::Role
    Properties:
      RoleName: BaseIAMRole
      AssumeRolePolicyDocument:
        Statement:
          - Action:
              - sts:AssumeRole
            Effect: Allow
            Principal:
              Service:
                - ec2.amazonaws.com
        Version: "2012-10-17"
      Path: /
```

Reply:
Definitely not admin powers since it should function as a sane default, and erroring out on something obvious before a person gets too far makes sense to me.